### ISSUE TITLE
dynamically or using php add tabulator options

### ISSUE THREAD

**Issue by JimNickerson** [olifolkerd/tabulator#1379]
I wonder if there is a way to insert the tabulator options like is done for columns and data ?
` tabulatorOptions = <?php

                           echo $_SESSION['tabulatorOptionsJason'];
                           //for example
                           //[
                           // {"height": "311px"},
                           // {"layout": "fitColumns"},
                           // {"placeholder": "No Dataavailable"},
                           // {"tooltips": "true"}
                           //]
                           ?>;

        $(document).ready(function () {
            //
            colData = <?php  echo $_SESSION['columDataJason']; ?>;
            tabledata = <?php  echo $_SESSION['tableDataJason']; ?>;
            //
            //create Tabulator on DOM element with id "data-table"
            $("#data-table").tabulator({
                height: tabulatorOptions[0],
                layout: "fitColumns",
                placeholder: "No Data Set",
                tooltips: true,
                index: "header",
                columns: colData,
                selectable: true,
                responsiveLayout: false,
                history: false,
                tooltips: function (cell) {
                    //cell - cell component
                    var cellValue = cell.getValue();
                    if (typeof (cellValue) == "undefined") {
                        cellValue = "";
                        return cellValue;// documentation says return string false
                    }
                    //function should return a string for the tooltip of false to hide the tooltip
                    //return cell.getColumn().getField() + " - " + cell.getValue(); //return cells "field - value";
                    return cell.getColumn().getField() + " - " + cellValue; //return cells "field - value";
                },
`

**Comment by Nyut0n** [comment]
Hey @JimNickerson,

I think you're looking for [json_encode](http://php.net/manual/en/function.json-encode.php).
You can of course create an options array in php and parse it as json object later.

```php
<?php
$tabulatorOptions = array(
  'height' => '311px',
  'layout' => 'fitColumns',
  'placeholder' => 'No data available',
  'tooltips' => TRUE,
  // More options, i.e. colums and data
);
?>
```
Then in JS:
```javascript
var tabulatorOptions = <?php echo json_encode($tabulatorOptions); ?>;
$("#data-table").tabulator(tabulatorOptions);
```

But remember that you can't stringify functions, so you'll have to add all the functions in the javascript after the json has parsed.
```javascript
var tabulatorOptions = <?php echo json_encode($tabulatorOptions); ?>;
tabulatorOptions.tooltips = function(cell) {
  /* function code */
};
$("#data-table").tabulator(tabulatorOptions);
```

**Comment by tomvanlier** [comment]
I don't get the question totally, but is this something you would like to do ?

Php :

$kolommenprofielenbeheer  = array();
$kolommenprofielbeheer = '{title:"Omschrijving", field:"omschrijvingprofiel",align:"center", formatter:"plaintext",editor:true,validator:"required",frozen:true},';

foreach($machtigingen as $key => $value){
                    
$kolommenprofielbeheer = $kolommenprofielbeheer.'{title:"'.$value.'", field:"'.$key.'", align:"center",formatter:"tickCross" , cellClick:function(e, cell){cell.setValue(!cell.getValue());}},';               
                                       
}

JS :

 var kolommenprofielbeheer = [<?php echo($kolommenprofielbeheer) ?>];

Tabulator :

columns: kolommenprofielbeheer

Basically I create a php string from an array with key-value pairs coming out of my db. That string gets converted to a JS-var to feed it into the tabulator. This allows for creating dynamical columns, which ofc all need to get the same formatter etc ... Variations on it can be made ofc, but using a loop will be really difficult if you don't want the same column style.

Let me know if it helped you, or if I got you wrong :)
